compiler/modules/front.syntax/parse.bal

```diff
@@ -185,11 +185,14 @@ function parseFunctionDefinition(Tokenizer tok, ModulePart part, Visibility vis,
     check tok.advance();
     Position namePos = tok.currentStartPos();
     string name = check tok.expectIdentifier();
-    string[] paramNames = [];
-    FunctionTypeDesc typeDesc = check parseFunctionTypeDesc(tok, paramNames);
+    FunctionTypeDesc typeDesc = check parseFunctionTypeDesc(tok, true);
     StmtBlock body = check parseStmtBlock(tok);
     Position endPos = tok.previousEndPos();
-    FunctionDefn defn = { startPos, endPos, name, vis, paramNames, typeDesc, namePos, body, part };
+    FunctionParam [] params = [];
```

Comment:
You shouldn't need to construct the array twice, because FunctionParam is a subtype of FunctionTypeParam.

Reply:
fixed due to passing `FunctionParams[]` to `parseFunctionTypeDesc`